gcloud-java-dns/src/main/java/com/google/gcloud/testing/LocalDnsHelper.java

```diff
@@ -0,0 +1,1426 @@
+   */
+  Response getZone(String projectId, String zoneName, String[] fields) {
+    ZoneContainer container = findZone(projectId, zoneName);
+    if (container == null) {
+      return Error.NOT_FOUND.response(String.format(
+          "The 'parameters.managedZone' resource named '%s' does not exist.", zoneName));
+    }
+    ManagedZone result = OptionParsersAndExtractors.extractFields(container.zone(), fields);
+    try {
+      return new Response(HTTP_OK, jsonFactory.toString(result));
+    } catch (IOException e) {
+      return Error.INTERNAL_ERROR.response(String.format(
+          "Error when serializing managed zone %s in project %s.", zoneName, projectId));
+    }
+  }
+
+  /**
+   * We assume that every project exists. If we do not have it in the collection yet, we just create
+   * a new default project instance with default quota.
+   */
+  Response getProject(String projectId, String[] fields) {
+    ProjectContainer defaultProject = createProject(projectId);
+    projects.putIfAbsent(projectId, defaultProject);
+    Project project = projects.get(projectId).project(); // project is now guaranteed to exist
+    Project result = OptionParsersAndExtractors.extractFields(project, fields);
+    try {
+      return new Response(HTTP_OK, jsonFactory.toString(result));
+    } catch (IOException e) {
+      return Error.INTERNAL_ERROR.response(
+          String.format("Error when serializing project %s.", projectId));
+    }
+  }
+
+  /**
+   * Creates a project. It generates a project number randomly (we do not have project numbers
+   * available).
+   */
+  private ProjectContainer createProject(String projectId) {
+    Quota quota = new Quota();
+    quota.setManagedZones(10000);
+    quota.setRrsetsPerManagedZone(10000);
+    quota.setRrsetAdditionsPerChange(100);
+    quota.setRrsetDeletionsPerChange(100);
+    quota.setTotalRrdataSizePerChange(10000);
+    quota.setResourceRecordsPerRrset(100);
+    Project project = new Project();
+    project.setId(projectId);
+    project.setNumber(new BigInteger(String.valueOf(
+        Math.abs(ID_GENERATOR.nextLong() % Long.MAX_VALUE))));
+    project.setQuota(quota);
+    return new ProjectContainer(project);
+  }
+
+  /**
+   * Deletes a zone.
+   */
+  Response deleteZone(String projectId, String zoneName) {
+    ProjectContainer projectContainer = projects.get(projectId);
+    ZoneContainer previous = projectContainer.zones.remove(zoneName);
+    return previous == null
+        // map was not in the collection
+        ? Error.NOT_FOUND.response(String.format(
+        "The 'parameters.managedZone' resource named '%s' does not exist.", zoneName))
+        : new Response(HTTP_NO_CONTENT, "{}");
+  }
+
+  /**
+   * Creates new managed zone and stores it in the collection. Assumes that project exists.
+   */
+  Response createZone(String projectId, ManagedZone zone, String[] fields) {
+    checkNotNull(zone, "Zone to create cannot be null");
```

Comment:
Is it possible for the user to create a request such that we get to createZone with a null zone?  If not, I'd remove this check.  If so, I would suggest returning a error response so that the user can see the error message.